### PR TITLE
HW-Lesson_5

### PR DESCRIPTION
**Pull request by annahappy961**

### CODE REVIEW

File: features/environment.py [hunk 1]
```diff
@@ -14,8 +14,8 @@ def browser_init(context):
 
     context.driver.maximize_window()
 
-    # context.driver.implicitly_wait(10)
-    context.driver.wait = WebDriverWait(context.driver, 10)
+    context.driver.implicitly_wait(10)
+    context.wait = WebDriverWait(context.driver, 10)
```
Comment (la-lev) [inline features/environment.py:18]: 👍

**Review by la-lev — COMMENTED** [review]
LGTM 👍

File: features/tests/search_page_ui.feature [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Feature: Test for Search page UI
+
+  Scenario: Verify each product has an image amd a name
+    Given Open Target main page
+    When Search for candy
+    Then Verify 29 images in shown
+    And Verify 29 names in shown
```
Comment (la-lev) [inline features/tests/search_page_ui.feature:7]: Great!

File: features/steps/search_results_page_steps.py [hunk 3]
```diff
@@ -58,3 +70,34 @@ def store_product_names(context):
...
+@then("Verify {amount} names in shown")
+def verify_product_name_shown(context, amount):
+    expected_names = int(amount)
+    scroll_and_verify_elements(context, PRODUCT_NAME, expected_names)
+
+
+@then("Verify {amount} images in shown")
+def verify_product_name_shown(context, amount):
+    expected_amount = int(amount)
+    scroll_and_verify_elements(context, PRODUCT_IMG, expected_amount)
```
Comment (la-lev) [inline features/steps/search_results_page_steps.py:103]: This should work 👍

File: features/steps/search_results_page_steps.py [hunk 4]
```diff
@@ -24,21 +28,28 @@ def verify_search_results_page_url(context, expected_part_url):
...
 
 
+@when("Store product price")
+def store_product_price(context):
+    context.product_price = context.driver.find_element(*PRODUCT_PRICE).text
+
+
 @when('Close side navigation')
 def side_nav_click_add_to_cart(context):
+    sleep(10)
```
Comment (la-lev) [inline features/steps/search_results_page_steps.py:52]: you can probably wait until SIDE_NAV_PREV_BUTTON is clickable here

File: features/environment.py (hunk 1 above)
Comment (annahappy961) [reply features/environment.py:18]: Thanks!

File: features/steps/search_results_page_steps.py (hunk 4 above)
Comment (annahappy961) [reply features/steps/search_results_page_steps.py:52]: I tried to use wait here. It didn't work for me. Not sure why.

**Pull request merged**